statefun-flink/statefun-flink-extensions/src/main/java/org/apache/flink/statefun/extensions/ComponentJsonObject.java

```diff
@@ -53,7 +53,7 @@
   private final ObjectNode rawObjectNode;
 
   private final TypeName binderTypename;
-  private final JsonNode specJsonNode;
```

Comment:
[Major] I'd personally keep this class immutable. It's easier to maintain and easier to reason about. Why not just create a new ComponentJsonObject when you need a new one?